presto-hive/src/test/java/com/facebook/presto/hive/parquet/ParquetTester.java

```
@@ -171,49 +210,125 @@ public void assertRoundTrip(ObjectInspector objectInspector, Iterable<?> writeVa
+                    }
+                }
+            }
+            assertFalse(iterator.hasNext());
+        }
+    }
 
-        long size = fileSystem.getFileStatus(path).getLen();
-        FSDataInputStream inputStream = fileSystem.open(path);
-        ParquetDataSource dataSource = new HdfsParquetDataSource(path, size, inputStream);
+    private static Object getActualCursorValue(RecordCursor cursor, Type type)
+            throws IOException
+    {
+        Object fieldFromCursor = getFieldFromCursor(cursor, type, 0);
+        if (fieldFromCursor == null) {
+            return null;
+        }
+        if (isStructuralType(type)) {
+            Block block = (Block) fieldFromCursor;
+            if (isArrayType(type)) {
+                Type elementType = ((ArrayType) type).getElementType();
+                return toArrayValue(block, elementType);
+            }
+            else if (isMapType(type)) {
+                MapType mapType = (MapType) type;
+                return toMapValue(block, mapType.getKeyType(), mapType.getValueType());
+            }
+            else if (isRowType(type)) {
+                return toRowValue(block, type.getTypeParameters());
+            }
+        }
+        if (type instanceof DecimalType) {
+            DecimalType decimalType = (DecimalType) type;
+            return new SqlDecimal((BigInteger) fieldFromCursor, decimalType.getPrecision(), decimalType.getScale());
+        }
+        if (isVarcharType(type)) {
+            return new String(((Slice) fieldFromCursor).getBytes());
+        }
+        if (VARBINARY.equals(type)) {
+            return new SqlVarbinary(((Slice) fieldFromCursor).getBytes());
+        }
+        if (DateType.DATE.equals(type)) {
+            return new SqlDate(((Long) fieldFromCursor).intValue());
+        }
+        if (TimestampType.TIMESTAMP.equals(type)) {
+            return new SqlTimestamp((long) fieldFromCursor, UTC_KEY);
+        }
+        return fieldFromCursor;
```

Is it possible to hit this return statement? If it's hit this means we are not handling some type properly and we silently return an `Object` for some `type`. Can we throw a runtime exception here instead of returning?

---

This method relies on getFieldFromCursor(cursor, type, 0) type checks. It's purpose is to cast some Hive cursor types to corresponding Java types which were used to create parquet file.